core/trino-main/src/main/java/io/trino/execution/SqlQueryExecution.java

```diff
@@ -365,8 +365,8 @@ public BasicQueryInfo getBasicQueryInfo()
     @Override
     public void start()
     {
-        try (SetThreadName ignored = new SetThreadName("Query-%s", stateMachine.getQueryId())) {
-            try {
+        try {
+            try (SetThreadName ignored = new SetThreadName("Query-%s", stateMachine.getQueryId())) {
```

Comment:
Why change this?  Also this seems unrelated, so I'd move it to a different commit if it is needed

Reply:
Previously, if it throws an exception(regardless of likelihood), previously it was caught by LocalDispatchQuery since `start()` was synchronous. However, after this change which makes `start()` asynchronous, the exception will not be caught properly, so the query will not be failed properly. If you want, I will move this to a separate commit. Or do you want to have a separate PR?

Reply:
Add a comment explaining this